tests/core/commands/test_init.py

```diff
@@ -193,6 +198,252 @@ def test_template_filename(local_client):
 
         (local_client.path / ".renku").mkdir()
 
-        create_from_template(template_folder, local_client, name="test")
+        create_from_template(template_folder, local_client, name="test", metadata=template["default_metadata"])
 
         assert (local_client.path / "test.r").exists()
+
+
+def test_update_from_template(local_client, mocker, template):
+    """Test repository update from a template."""
+    with TemporaryDirectory() as tempdir:
+        temppath = Path(tempdir) / "local"
+        shutil.copytree(str(template_local), str(temppath))
+        manifest = read_template_manifest(temppath)
+        template_path = temppath / manifest[0]["folder"]
+        create_from_template(
+            template_path,
+            local_client,
+            "name",
+            {**template["default_metadata"], **template["metadata"]},
+            template_version="0.0.1",
+            automated_update=True,
+        )
+        project_files = [
+            f
+            for f in local_client.path.glob("**/*")
+            if ".git" not in str(f)
+            and not str(f).endswith(".renku/metadata.yml")
+            and not str(f).endswith(".renku/template_checksums.json")
+        ]
+        template_files = []
+        for project_file in project_files:
+            expected_file = template_path / project_file.relative_to(local_client.path)
+            template_files.append(expected_file)
+            assert expected_file.exists()
+
+        # NOTE: Update from template
+        fetch_template = mocker.patch("renku.core.commands.init.fetch_template")
+        fetch_template.return_value = (manifest, temppath, "renku", "0.0.2")
```

Comment:
It's better to factor the common setup code for this test and the following in a separate function to make the tests shorter and more readable.

Reply:
Well, each setup code is different, but I tried to hide it away in a function as a fixture.